openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -205,3 +213,88 @@ def award_program_certificates(self, username):
+            'mode': certificate.mode,
+            'type': COURSE_CERTIFICATE,
+        },
+        'attributes': [
+            {
+                'name': 'visible_date',
+                'value': visible_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+            }
+        ]
+    })
+
+
+@task(bind=True, ignore_result=True, routing_key=ROUTING_KEY)
+def award_course_certificate(self, username, course_run_key):
+    """
+    This task is designed to be called whenever a student GeneratedCertificate is updated.
+    It can be called independently for a username and a course_run, but is invoked on each GeneratedCertificate.save.
+    """
+
+    LOGGER.info('Running task award_course_certificate for username %s', username)
+
+    countdown = 2 ** self.request.retries
+
+    # If the credentials config model is disabled for this
+    # feature, it may indicate a condition where processing of such tasks
+    # has been temporarily disabled.  Since this is a recoverable situation,
+    # mark this task for retry instead of failing it altogether.
+
+    if not CredentialsApiConfig.current().is_learner_issuance_enabled:
+        LOGGER.warning(
+            'Task award_course_certificate cannot be executed when credentials issuance is disabled in API config',
+        )
+        raise self.retry(countdown=countdown, max_retries=MAX_RETRIES)
+
+    try:
+        try:
+            user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            LOGGER.exception('Task award_course_certificate was called with invalid username %s', username)
+            # Don't retry for this case - just conclude the task.
+            return
+        # Get the cert for the course key and username if it's both passing and available in professional/verified
+        try:
+            certificate = GeneratedCertificate.eligible_certificates.get(
+                user=user.id,
+                course_id=course_run_key
+            )
+        except GeneratedCertificate.DoesNotExist:
+            LOGGER.exception(
+                'Task award_course_certificate was called without Certificate found for %s to user %s',
+                course_run_key,
+                username
+            )
+            return
+        if certificate.mode in CourseMode.VERIFIED_MODES + CourseMode.CREDIT_MODES:
+            try:
+                course_overview = CourseOverview.get_from_id(course_run_key)
+            except (CourseOverview.DoesNotExist, IOError):
+                LOGGER.exception(
+                    'Task award_course_certificate was called without course overview data for course %s',
+                    course_run_key
+                )
+                return
+            credentials_client = get_api_client(
+                CredentialsApiConfig.current(),
+                User.objects.get(username=settings.CREDENTIALS_SERVICE_USERNAME)
+            )
+            visible_date = display_date_for_certificate(course_overview, certificate)
```

Comment:
Nice find.

Reply:
Hah yup, always a method to do what I want... just never know what it's called or where it is